Content.Server/GameObjects/Components/AdvertiseComponent.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Content.Server.Advertisements;
+using Content.Server.Interfaces.Chat;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.GameObjects.Components
+{
+    [RegisterComponent]
+    public class AdvertiseComponent : Component
+    {
+        [Dependency] private readonly IChatManager _chatManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+        public override string Name => "Advertise";
+
+        private CancellationTokenSource _cancellationSource = new();
+
+        /// <summary>
+        /// Minimum time to wait before saying a new ad, in ms.
+        /// </summary>
+        [field: DataField("minWait")]
+        private int MinWait { get; } = 40000;
+
+        /// <summary>
+        /// Maximum time to wait before saying a new ad, in ms.
+        /// </summary>
+        [field: DataField("maxWait")]
+        private int MaxWait { get; } = 80000;
+
+        [field: DataField("pack")]
+        private string PackPrototypeId { get; } = string.Empty;
+
+        private List<string> _advertisements = new();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _prototypeManager.TryIndex(PackPrototypeId, out AdvertisementsPackPrototype? packPrototype);
+
+            // Load advertisements pack
+            if (string.IsNullOrEmpty(PackPrototypeId) || packPrototype == null)
+            {
+                // If there is no pack, log a warning and remove the component
+                Logger.Log(LogLevel.Warning, Owner.Name + " has " + Name + "Component but no advertisments pack.");
```

Comment:
```suggestion
                Logger.Warning($"{Owner} has {Name} Component but no advertisments pack.");
```

Logging `Owner.Name` is pretty crappy because it doesn't have good debugging info like prototype ID. Just logging the entity's `ToString()` (what my suggestion does) DOES print prototype ID so makes debugging easier.